http/src/response/mod.rs

```diff
@@ -233,22 +238,49 @@ impl<T> Response<T> {
     ///
     /// [`text`]: Self::text
     pub fn bytes(self) -> BytesFuture {
-        let body = self.inner.into_body();
+        #[cfg(feature = "compression")]
+        let compressed = self
+            .inner
+            .headers()
+            .get(hyper::header::CONTENT_ENCODING)
+            .is_some();
 
-        // We need to do additional processing here due to the return type of
-        // `body::aggregate` being `impl Buf`.
-        let fut = async {
-            let mut aggregate = body::aggregate(body).await?;
+        let body = self.inner.into_body();
 
-            // Create a buffer filled with zeroes so we can copy the aggregate
-            // body into it.
-            //
-            // Using `vec!` is the fastest way to do this, despite it being a
-            // macro and having unsafe internals.
-            let mut buf = vec![0; aggregate.remaining()];
-            aggregate.copy_to_slice(&mut buf);
+        let fut = async move {
+            {
+                #[cfg(feature = "compression")]
+                if compressed {
+                    use hyper::body::Buf;
+                    use std::io::Read;
+
+                    let mut buf = Vec::with_capacity(256);
+                    brotli::Decompressor::new(
+                        hyper::body::aggregate(body)
```

Comment:
```suggestion
                        body::aggregate(body)
```
We already import `body` above, so let's use it